From 3467c559d82943f7339578f92b0817b63b8b7ddc Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Tue, 4 Jul 2017 08:15:18 +0800
Subject: [PATCH] ARM: dts: rockchip: rk3288 re-name dts files

Intent to keep same dts with upstream version, this patch does:
1. restore rk3288-evb-act8846.dts, keep without android configurations.
2. add char of "-android-" to android dts files.

Change-Id: Ic063f1187160f88e9ec8a779749ff7b17cfac9e8
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm/boot/dts/Makefile                    |   6 +-
 ...act8846-edp.dts => rk3288-evb-act8846.dts} | 176 +++++++++++++-----
 ...p.dts => rk3288-evb-android-rk818-edp.dts} |   2 +-
 ....dts => rk3288-evb-android-rk818-mipi.dts} |   2 +-
 4 files changed, 138 insertions(+), 48 deletions(-)
 rename arch/arm/boot/dts/{rk3288-evb-act8846-edp.dts => rk3288-evb-act8846.dts} (73%)
 rename arch/arm/boot/dts/{rk3288-evb-rk818-edp.dts => rk3288-evb-android-rk818-edp.dts} (99%)
 rename arch/arm/boot/dts/{rk3288-evb-rk818-mipi.dts => rk3288-evb-android-rk818-mipi.dts} (99%)

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 872f0cb1a610..c5397ece3b5f 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -522,10 +522,10 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += \
 	rk3229-echo-v10.dtb \
 	rk3229-evb.dtb \
 	rk3229-gva-sdk.dtb \
-	rk3288-evb-act8846-edp.dtb \
+	rk3288-evb-act8846.dtb \
+	rk3288-evb-android-rk818-edp.dtb \
+	rk3288-evb-android-rk818-mipi.dtb \
 	rk3288-evb-rk808.dtb \
-	rk3288-evb-rk818-edp.dtb \
-	rk3288-evb-rk818-mipi.dtb \
 	rk3288-fennec.dtb \
 	rk3288-firefly-beta.dtb \
 	rk3288-firefly.dtb \
diff --git a/arch/arm/boot/dts/rk3288-evb-act8846-edp.dts b/arch/arm/boot/dts/rk3288-evb-act8846.dts
similarity index 73%
rename from arch/arm/boot/dts/rk3288-evb-act8846-edp.dts
rename to arch/arm/boot/dts/rk3288-evb-act8846.dts
index 5f5fb3277bb5..307c03fecca9 100644
--- a/arch/arm/boot/dts/rk3288-evb-act8846-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-act8846.dts
@@ -40,11 +40,109 @@
 
 /dts-v1/;
 #include "rk3288-evb.dtsi"
-#include "rk3288-android.dtsi"
 
 / {
 	compatible = "rockchip,rk3288-evb-act8846", "rockchip,rk3288";
 
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		brightness-levels = <
+			  0   1   2   3   4   5   6   7
+			  8   9  10  11  12  13  14  15
+			 16  17  18  19  20  21  22  23
+			 24  25  26  27  28  29  30  31
+			 32  33  34  35  36  37  38  39
+			 40  41  42  43  44  45  46  47
+			 48  49  50  51  52  53  54  55
+			 56  57  58  59  60  61  62  63
+			 64  65  66  67  68  69  70  71
+			 72  73  74  75  76  77  78  79
+			 80  81  82  83  84  85  86  87
+			 88  89  90  91  92  93  94  95
+			 96  97  98  99 100 101 102 103
+			104 105 106 107 108 109 110 111
+			112 113 114 115 116 117 118 119
+			120 121 122 123 124 125 126 127
+			128 129 130 131 132 133 134 135
+			136 137 138 139 140 141 142 143
+			144 145 146 147 148 149 150 151
+			152 153 154 155 156 157 158 159
+			160 161 162 163 164 165 166 167
+			168 169 170 171 172 173 174 175
+			176 177 178 179 180 181 182 183
+			184 185 186 187 188 189 190 191
+			192 193 194 195 196 197 198 199
+			200 201 202 203 204 205 206 207
+			208 209 210 211 212 213 214 215
+			216 217 218 219 220 221 222 223
+			224 225 226 227 228 229 230 231
+			232 233 234 235 236 237 238 239
+			240 241 242 243 244 245 246 247
+			248 249 250 251 252 253 254 255>;
+		default-brightness-level = <128>;
+		enable-gpios = <&gpio7 2 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&bl_en>;
+		pwms = <&pwm0 0 1000000 PWM_POLARITY_INVERTED>;
+	};
+
+	adc-keys {
+		compatible = "adc-keys";
+		io-channels = <&saradc 1>;
+		io-channel-names = "buttons";
+		poll-interval = <100>;
+		keyup-threshold-microvolt = <1800000>;
+
+		button-up {
+			label = "Volume Up";
+			linux,code = <KEY_VOLUMEUP>;
+			press-threshold-microvolt = <100000>;
+		};
+
+		button-down {
+			label = "Volume Down";
+			linux,code = <KEY_VOLUMEDOWN>;
+			press-threshold-microvolt = <300000>;
+		};
+
+		menu {
+			label = "Menu";
+			linux,code = <KEY_MENU>;
+			press-threshold-microvolt = <640000>;
+		};
+
+		esc {
+			label = "Esc";
+			linux,code = <KEY_ESC>;
+			press-threshold-microvolt = <1000000>;
+		};
+
+		home  {
+			label = "Home";
+			linux,code = <KEY_HOME>;
+			press-threshold-microvolt = <1300000>;
+		};
+	};
+
+	gpio-keys {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		autorepeat;
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&pwrbtn>;
+
+		button@0 {
+			gpios = <&gpio0 5 GPIO_ACTIVE_LOW>;
+			linux,code = <116>;
+			label = "GPIO Key Power";
+			linux,input-type = <1>;
+			gpio-key,wakeup = <1>;
+			debounce-interval = <100>;
+		};
+	};
+
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
 		clocks = <&hym8563>;
@@ -89,15 +187,24 @@
 &edp {
 	vcc-supply = <&vdd10_lcd>;
 	vccio-supply = <&vcc18_lcd>;
+	status = "okay";
 };
 
-&gpu {
+&edp_phy {
 	status = "okay";
-	mali-supply = <&vdd_gpu>;
 };
 
-&hdmi_analog_sound {
+&edp_panel {
+	status = "okay";
+	compatible ="lg,lp079qx1-sp0v", "simple-panel";
+	backlight = <&backlight>;
+	enable-gpios = <&gpio7 4 GPIO_ACTIVE_HIGH>;
+	pinctrl-0 = <&lcd_cs>;
+};
+
+&gpu {
 	status = "okay";
+	mali-supply = <&vdd_gpu>;
 };
 
 &i2c0 {
@@ -250,43 +357,6 @@
 	};
 };
 
-&edp {
-	status = "okay";
-};
-
-&edp_panel {
-	compatible ="lg,lp079qx1-sp0v", "simple-panel";
-	enable-gpios = <&gpio7 4 GPIO_ACTIVE_HIGH>;
-	pinctrl-0 = <&lcd_cs>;
-	power-supply = <&vcc_lcd>;
-	delay,prepare = <120>;
-	status = "okay";
-
-	disp_timings: display-timings {
-		native-mode = <&timing0>;
-
-		timing0: timing0 {
-			clock-frequency = <200000000>;
-			hactive = <1536>;
-			vactive = <2048>;
-			hfront-porch = <12>;
-			hsync-len = <16>;
-			hback-porch = <48>;
-			vfront-porch = <8>;
-			vsync-len = <4>;
-			vback-porch = <8>;
-			hsync-active = <0>;
-			vsync-active = <0>;
-			de-active = <0>;
-			pixelclk-active = <0>;
-		};
-	};
-};
-
-&edp_phy {
-	status = "okay";
-};
-
 &i2c1 {
 	status = "okay";
 	clock-frequency = <400000>;
@@ -309,11 +379,31 @@
 	};
 };
 
-&lvds_panel {
-	power-supply = <&vcc_lcd>;
+&rga {
+	status = "okay";
+};
+
+&sound {
+	status = "okay";
+};
+
+&uart2 {
+	status = "okay";
 };
 
 &pinctrl {
+	backlight {
+		bl_en: bl-en {
+			rockchip,pins = <7 2 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+	};
+
+	buttons {
+		pwrbtn: pwrbtn {
+			rockchip,pins = <0 5 RK_FUNC_GPIO &pcfg_pull_up>;
+		};
+	};
+
 	lcd {
 		lcd_en: lcd-en  {
 			rockchip,pins = <7 3 RK_FUNC_GPIO &pcfg_pull_none>;
diff --git a/arch/arm/boot/dts/rk3288-evb-rk818-edp.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts
similarity index 99%
rename from arch/arm/boot/dts/rk3288-evb-rk818-edp.dts
rename to arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts
index abdd2ffc5196..e31c99e248a4 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk818-edp.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-edp.dts
@@ -45,7 +45,7 @@
 #include "rk3288-android.dtsi"
 
 / {
-	compatible = "rockchip,rk3288-evb-rk818", "rockchip,rk3288";
+	compatible = "rockchip,rk3288-evb-android-rk818", "rockchip,rk3288";
 
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
diff --git a/arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts b/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts
similarity index 99%
rename from arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts
rename to arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts
index 11c3c4c0cac8..24c916d0ff5c 100644
--- a/arch/arm/boot/dts/rk3288-evb-rk818-mipi.dts
+++ b/arch/arm/boot/dts/rk3288-evb-android-rk818-mipi.dts
@@ -45,7 +45,7 @@
 #include "rk3288-android.dtsi"
 
 / {
-	compatible = "rockchip,rk3288-evb-rk818", "rockchip,rk3288";
+	compatible = "rockchip,rk3288-evb-android-rk818", "rockchip,rk3288";
 
 	sdio_pwrseq: sdio-pwrseq {
 		compatible = "mmc-pwrseq-simple";
-- 
2.35.3

